### ISSUE TITLE
A non-indexed optional array of objects can't have null as value for a key, raises a confusing error.

### ISSUE THREAD

**Issue by iamarora** [typesense/typesense#1030]
## Description

A non-indexed optional array of objects can't have null as value for a key, raises error.

### Schema
```
schema = {
    "enable_nested_fields": True,
    "fields": [
        {
            "name": "colleges",
            "type": "object[]",
            "index": False,
            "optional": True,
        }
    ]
}
```

## Expected Behavior
Should be able to insert below JSON, with one "degree" as null as it's optional.
```
{"colleges": [{
		"id": 1,
		"college": "Fort Hamilton",
		"degree": null
	}, {
		"id": 2,
		"college": "Brooklyn College",
		"degree": "Associate's degree"
	}, {
		"id": 3,
		"college": "Kingsborough Community College",
		"degree": "Bachelor of Business Administration - BBA"
	}]
}
```
I'm able to create/upsert using a blank string instead of null.

## Actual Behavior
With the null for degree I get the error:

Field `colleges_display.degree` must be an array of string.

This was confusing and took a long time to debug. Went down the path of defining nested arrays in a wrong manner. 

## Metadata

**Typesense Version**: V0.24.1

**OS**: Using Typesense cloud.

**Comment by kishorenc** [comment]
@iamarora 

I believe we've fixed this in recent builds. Let me verify this and get back to you soon.

**Comment by kishorenc** [comment]
@iamarora 

Yes, this is fixed in recent 0.25 RC builds.

**Comment by iamarora** [comment]
> @iamarora
> 
> Yes, this is fixed in recent 0.25 RC builds.

Thanks for confirming! Feel free to close this issue out. Waiting for new releases with all the optimization on search and facets. :)
